arangod/VocBase/LogicalDataSource.h

```diff
@@ -124,7 +104,7 @@ class LogicalDataSource {
 
   virtual ~LogicalDataSource() = default;
 
-  Category const& category() const noexcept { return _category; }
+  virtual Category category() const noexcept = 0;
```

Comment:
let's just store Category as a member to avoid unnecessary virtual calls

Reply:
That would also work, but I dislike that approach for the reason that then in theory it would be possible to create a LogicalView with category `kCollection`. I prefer the category to be part of the derived classes, so it's error-proof to use (and one less data member to carry around and think about). Of course this approach has the virtual calls as a downside, but I wouldn't expect any actual performance regression in our tests. Normally the category is determined once or maybe twice per LogicalDataSource for a high-level operation. E.g. an AQL query iterating over the docs in a collection will determine the category once or twice at start, but the actual query execution will not do it. 
So it shouldn't really matter IMO.

Reply:
@jsteemann we can add an assertion in `LogicalCollection`/`LogicalView` to ensure that category is correct